src/System.Net.Security/tests/FunctionalTests/NegotiateStreamInvalidOperationTest.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Net.Test.Common;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Net.Security.Tests
+{
+    [PlatformSpecific(TestPlatforms.Windows)] // NegotiateStream only supports client-side functionality on Unix
+    public class NegotiateStreamInvalidOperationTest
+    {
+        private readonly byte[] _sampleMsg = Encoding.UTF8.GetBytes("Sample Test Message");
+
+        [Fact]
+        public async Task NegotiateStream_StreamContractTest_Success()
+        {
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var client = new NegotiateStream(clientStream))
+            using (var server = new NegotiateStream(serverStream))
+            {
+                Assert.False(client.CanSeek);
+                Assert.False(client.CanRead);
+                Assert.False(client.CanTimeout);
+                Assert.False(client.CanWrite);
+                Assert.False(server.CanSeek);
+                Assert.False(server.CanRead);
+                Assert.False(server.CanTimeout);
+                Assert.False(server.CanWrite);
+
+                Assert.Throws<InvalidOperationException>(() => client.ReadTimeout);
+                Assert.Throws<InvalidOperationException>(() => client.WriteTimeout);
+                Assert.Throws<NotImplementedException>(() => client.Length);
+                Assert.Throws<NotImplementedException>(() => client.Position);
+                Assert.Throws<NotSupportedException>(() => client.Seek(0, new SeekOrigin()));
+
+                Task[] auth = new Task[2];
+                auth[0] = client.AuthenticateAsClientAsync();
+                auth[1] = server.AuthenticateAsServerAsync();
+
+                await TestConfiguration.WhenAllOrAnyFailedWithTimeout(auth);
```

Comment:
Nit: you could tighten this up if you wanted, to:
```C#
await TestConfiguration.WhenAllOrAnyFailedWithTimeout(
    client.AuthenticateAsClientAsync(),
    server.AuthenticateAsServerAsync());
```

Reply:
Fixed.